### DISCUSSION TITLE
FreeType bindings?

### DISCUSSION THREAD

**Discussion by desiwko** [dotnet/Silk.NET#777] [Ideas] [answered]
Are there any plans or thoughts on adding FreeType bindings as part of Silk.NET?  I think it would be a solid add.

I personally have been using a different C# FreeType binding, but I find it very much lacking in terms of integration and syntax compared to how Silk.NET handles bindings in general, and there doesn't seem to be much active work for something that seems useful across multiple graphics APIs as much as FreeType.

**Comment by HurricanKai** [comment, marked as answer]
We have generally recommended https://github.com/SixLabors/Fonts
Is there a reason FreeType bindings would be preferable?

**Reply by desiwko** [reply]
There's probably no technical reason I can think of other than I personally didn't know about SixLabors Font API before your response.  I can only use my experience as background - coming from a C++ background with OpenGL, nearly every resource out there has used FreeType as the defacto C and C++ font handling library.  Pretty much every OpenGL example dealing with fonts such as LearnOpenGL has historically used it, hence my suggestion as I assumed it was "standard".

As a C standard library I also figured FreeType fit well into the concept of what Silk.NET is trying to accomplish as well as how.

The only potential reason I can see against SixLabors at this point is that their Font API is still marked as BETA and the API already differs from the examples on their own website, even if minor.  Also, since you pointing out that it exists, I'm noticing that their Font API has some dependencies on additional SixLabors packages which otherwise may not be needed with FreeType.  Not sure if this is related to the BETA status or just something inherent to their design.

There may also be some incongruity in terms of licensing for various projects, but that's probably beyond the scope of this discussion at this point.

I figured it would be worth at least suggesting.  All said, now that I know about SixLabors, I'll give them a deeper looksie.

**Reply by HurricanKai** [reply]
I'll definitely put FreeType on the dev tracker, thanks for the suggestion!
